src/mysky/index.ts

```diff
@@ -0,0 +1,160 @@
+export type { CustomConnectorOptions } from "./connector";
+export { DacLibrary } from "./dac";
+
+import { PermCategory, PermHidden, Permission, PermType, PermWrite } from "skynet-interface-utils";
+
+import { Connector, CustomConnectorOptions } from "./connector";
+import { SkynetClient } from "../client";
+import { DacLibrary } from "./dac";
+import { RegistryEntry, setSignedEntry, SignedRegistryEntry } from "../registry";
+import { CustomGetJSONOptions, CustomSetJSONOptions, getOrCreateRegistryEntry, JsonData } from "../skydb";
+import { hexToUint8Array } from "../utils/string";
+import { Signature } from "../crypto";
+import { deriveDiscoverableTweak } from "./tweak";
+
+export async function loadMySky(
+  this: SkynetClient,
+  appPath: string,
+  customOptions?: CustomConnectorOptions
+): Promise<MySky> {
+  const mySky = await MySky.init(this, customOptions);
+  mySky.addPermissions(new Permission(appPath, PermHidden, PermWrite));
+
+  return mySky;
+}
+
+export const mySkyDomain = "skynet-mysky.hns";
+
+export class MySky {
+  // ============
+  // Constructors
+  // ============
+
+  constructor(protected connector: Connector, protected permissions: Permission[]) {}
+
+  static async init(client: SkynetClient, customOptions?: CustomConnectorOptions): Promise<MySky> {
+    const connector = await Connector.init(client, mySkyDomain, customOptions);
+
+    const domain = await client.extractDomain(window.location.hostname);
+    // TODO: Add requestor field to Permission?
+    // TODO: Are these permissions correct?
+    const perm = new Permission(domain, PermCategory.Hidden, PermType.Write);
+    const permissions = [perm];
+
+    return new MySky(connector, permissions);
+  }
+
+  // ==========
+  // Public API
+  // ==========
+
+  /**
+   * Loads the given DACs. Takes one or more instantiated DAC libraries.
+   */
+  async loadDacs(dac: DacLibrary, ...dacs: DacLibrary[]): Promise<void> {
+    if (dacs) {
+      dacs.unshift(dac);
+    } else {
+      dacs = [dac];
+    }
+
+    const promises: Promise<void>[] = [];
+    for (const dac of dacs) {
+      promises.push(this.loadDac(dac));
+    }
+
+    await Promise.all(promises);
+  }
+
+  async addPermissions(...permissions: Permission[]) {
```

Comment:
Why is this variadic? That's kind of weird. This will result in an array of arrays.

Reply:
Why can't we make it 

```
  async addPermissions(permissions: Permission[]) {
      this.permissions = [
            ....this.permissions,
            ....permissions,
      ]
  }
  ```

Reply:
No, I think that's right.

https://stackoverflow.com/questions/12699058/typescript-types-for-variadic-functions

Not sure why you say variadics are weird, this will be more user-friendly for devs in the majority of cases. If they have an array of permissions they can alsways do `mysky.addPermissions(...perms)`